src/colvarcomp_distances.cpp

```diff
@@ -1045,6 +1045,44 @@ colvar::rmsd::rmsd(std::string const &conf)
     // this is only required for ABF, but we do both groups here for better caching
     atoms->rot.request_group2_gradients(atoms->size());
   }
+
+  std::string perm_conf;
+  size_t pos = 0; // current position in config string
+  n_permutations = 1;
+
+  while (key_lookup(conf, "atomPermutation", &perm_conf, &pos)) {
+    std::vector<size_t> perm;
+    if (perm_conf.size()) {
+      std::istringstream is(perm_conf);
+      size_t index;
+      while (is >> index) {
+        std::vector<int> const &ids = atoms->ids();
+        size_t const ia = std::find(ids.begin(), ids.end(), index-1) - ids.begin();
+        if (ia == atoms->size()) {
+          cvm::error("Error: atom id " + cvm::to_str(index) +
+                    " is not a member of group \"atoms\".");
+          return;
+        }
+        if (std::find(perm.begin(), perm.end(), ia) != perm.end()) {
+          cvm::error("Error: atom id " + cvm::to_str(index) +
+                    " is mentioned more than once in atomPermutation list.");
+          return;
+        }
+        perm.push_back(ia);
+      }
+      if (perm.size() != atoms->size()) {
+        cvm::error("Error: symmetry permutation in input contains " + cvm::to_str(perm.size()) +
+                  " indices, but group \"atoms\" contains " + cvm::to_str(atoms->size()) + " atoms.");
+        return;
+      }
+      cvm::log("atomPermutation = " + cvm::to_str(perm));
+      n_permutations++;
+      // Record a copy of reference positions in new order
+      for (size_t ia = 0; ia < atoms->size(); ia++) {
```

Comment:
Remember about `-fno-for-scope`. We still need to remain compatible with Windows 98, man.

Reply:
Hey, you removed that flag in your patches! How am I supposed to catch that now???

Seriously though, this builds even with the flag, because the variable is not redefined after the loop. The previous instance is inside a block, not a for loop.

I've added the flag back into my build tree so I can catch the next instance of this.

Reply:
Fair enough. However, I've been checking that VMD builds and runs correctly using the pristine version from CVS. I think that's what we all should be doing, especially seeing as 99.99% of VMD users rely on precompiled builds.

I use the patches to enable the new features in the VMD build that I actually *use* myself in production.